.github/workflows/deploy-vercel.yml

```diff
@@ -0,0 +1,156 @@
+# Copyright 2025 The Multigres Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Deploy to Vercel
+
+env:
+  VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
+  VERCEL_PROJECT_ID: ${{ secrets.VERCEL_PROJECT_ID }}
+  VERCEL_PROTECTION_BYPASS: ${{ secrets.VERCEL_PROTECTION_BYPASS }}
+
+on:
+  push:
+    branches: [ main ]
+    paths:
+      - 'site/**'
+  pull_request:
+    branches: [ main ]
+    paths:
+      - 'site/**'
+    types: [ opened, synchronize, reopened ]
+
+jobs:
+  deploy-preview:
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-24.04
+    environment: ci
+    
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Validate Required Secrets
+      run: |
+        if [ -z "${{ secrets.VERCEL_ORG_ID }}" ]; then
+          echo "❌ VERCEL_ORG_ID secret is not configured"
+          echo "Please configure the VERCEL_ORG_ID secret in your repository settings"
+          exit 1
+        fi
+        
+        if [ -z "${{ secrets.VERCEL_PROJECT_ID }}" ]; then
+          echo "❌ VERCEL_PROJECT_ID secret is not configured"
+          echo "Please configure the VERCEL_PROJECT_ID secret in your repository settings"
+          exit 1
+        fi
+        
+        if [ -z "${{ secrets.VERCEL_PROTECTION_BYPASS }}" ]; then
+          echo "❌ VERCEL_PROTECTION_BYPASS secret is not configured"
+          echo "Please configure the VERCEL_PROTECTION_BYPASS secret in your repository settings"
+          echo "This should be the value from the x-vercel-protection-bypass header"
+          exit 1
+        fi
+        
+        echo "✅ All required secrets are configured"
+
+    - name: Install pnpm
+      uses: pnpm/action-setup@v3
+      with:
+        version: 9.15.5
+
+    - name: Setup Node.js
+      uses: actions/setup-node@v4
+      with:
+        node-version: '18'
```

Comment:
I think node 18 is EOL - https://nodejs.org/en/about/previous-releases#looking-for-the-latest-release-of-a-version-branch

Also, it would be better to use a `.nvmrc` file so that whenever you bump the version, you bump it at a single place.

Could be done using the `node-version-file` arg,